src/gallium/drivers/iris/iris_batch.c

```diff
@@ -936,16 +986,34 @@ submit_batch(struct iris_batch *batch)
     * in theory try to grab bo_deps_lock. Let's keep it safe and decode
     * outside the lock.
     */
-   if (INTEL_DEBUG(DEBUG_BATCH))
-      decode_batch(batch);
-
+   if (INTEL_DEBUG(DEBUG_BATCH)) {
+      if (getenv("CAPTURE_FRAME_NO") != NULL) {
+         if (getenv("FRAME_NO") != NULL && atoi(getenv("FRAME_NO")) == atoi(getenv("CAPTURE_FRAME_NO"))) {
+            FILE* logfile = create_log_file();
+            if (logfile) {
+               batch->decoder.fp = logfile;
+               decode_batch(batch);
+               close_log_file(&batch->decoder.fp);
+            }
+         }
+      } else {
+         decode_batch(batch);
+      }
+   }
    simple_mtx_lock(bo_deps_lock);
 
    update_batch_syncobjs(batch);
 
    if (INTEL_DEBUG(DEBUG_BATCH | DEBUG_SUBMIT)) {
-      dump_fence_list(batch);
-      dump_bo_list(batch);
+      if (getenv("CAPTURE_FRAME_NO") != NULL) {
+         if (getenv("FRAME_NO") != NULL && atoi(getenv("FRAME_NO")) == atoi(getenv("CAPTURE_FRAME_NO"))) {
+            dump_fence_list(batch);
+            dump_bo_list(batch);
+         }
+      } else {
+         dump_fence_list(batch);
+         dump_bo_list(batch);
+      }
```

Comment:
It is better to combine the check, for example:
if (getenv("CAPTURE_FRAME_NO") == NULL ||
   (getenv("CAPTURE_FRAME_NO") != NULL && getenv("FRAME_NO") != NULL && atoi(getenv("FRAME_NO")) == atoi(getenv("CAPTURE_FRAME_NO")))